packages/new-editor/src/components/editor/variables/detail/Variable.tsx

```diff
@@ -1,34 +1,49 @@
 import { Fieldset, Input, Message, SimpleSelect, Textarea } from '@axonivy/ui-components';
+import { getVariable } from '../../../../data/data';
+import type { TreePath } from '../../../../types/config';
 import { metadataOptions, type Variable } from '../../data/Variable';
 
 type VariableProps = {
-  variable?: Variable;
+  variables: Array<Variable>;
+  variablePath: TreePath;
+  setVariables: (variables: Array<Variable>) => void;
 };
 
-export const VariableDetail = ({ variable }: VariableProps) => {
+export const VariableDetail = ({ variables, variablePath, setVariables }: VariableProps) => {
+  const variable = getVariable(variables, variablePath);
   if (!variable) {
     return <Message>Select a variable to edit.</Message>;
   }
 
   const selectedMetadataOption = metadataOptions.find(option => option.value === variable.metadata);
   const doesNotHaveChildren = variable.children.length == 0;
 
+  const handleVariableAttributeChange = (key: keyof Variable, value: any) => {
+    const newVariables = structuredClone(variables);
+    getVariable(newVariables, variablePath)![key] = value;
+    setVariables(newVariables);
+  };
+
   return (
     <>
       <Fieldset label='Name'>
-        <Input value={variable.name} />
+        <Input value={variable.name} onChange={event => handleVariableAttributeChange('name', event.target.value)} />
```

Comment:
Refactoring the attribute name in the interface `Variable` will not change the string that is used to identify which attribute to change. TypeScript will however still show an error afterwards. Any way to make this work without having to create a function for each attribute?

Reply:
Not sure what you mean here

Reply:
The attribute to be updated is identified by a string (the name of the attribute). If I refactor the name of the attribute, this string does not change. I would like this string to update as well with the refactor.

![refactoring-issue](https://github.com/axonivy/config-editor-client/assets/141232142/f53334a8-9698-4abf-80c3-c8a3af2e06f4)